dids/diddht/diddht.go

```diff
@@ -1,175 +1,252 @@
 package diddht
 
 import (
+	"context"
 	"errors"
 	"fmt"
+	"net/http"
 	"strings"
+	"sync"
+	"time"
 
-	"golang.org/x/net/dns/dnsmessage"
-
+	"github.com/tbd54566975/web5-go/crypto"
+	"github.com/tbd54566975/web5-go/crypto/dsa"
+	"github.com/tbd54566975/web5-go/dids/did"
 	"github.com/tbd54566975/web5-go/dids/didcore"
+	"github.com/tbd54566975/web5-go/dids/diddht/internal/bep44"
+	"github.com/tbd54566975/web5-go/dids/diddht/internal/dns"
+	"github.com/tbd54566975/web5-go/dids/diddht/internal/pkarr"
+	"github.com/tv42/zbase32"
 )
 
-// Decoder is used to structure the DNS representation of a DID
-type Decoder struct {
-	rootRecord string
-	records    map[string]string
+// relay is the internal interface used to publish Pakrr messages to the DHT
+type relay interface {
+	Put(didID string, payload *bep44.Message) error
+	PutWithContext(ctx context.Context, didID string, payload *bep44.Message) error
+
+	Fetch(didID string) (*bep44.Message, error)
+	FetchWithContext(ctx context.Context, didID string) (*bep44.Message, error)
+}
+
+var defaultRelay relay
+var once sync.Once
+
+// getDefaultRelay returns the default Pkarr relay client.
+func getDefaultRelay() relay {
+	once.Do(func() {
+		defaultRelay = pkarr.NewClient("", http.DefaultClient)
+	})
+
+	return defaultRelay
+}
+
+// CreateOption is the type returned from each individual option function
+type CreateOption func(*createOptions)
```

Comment:
@mistermoe these are roughly the same options that did:web uses, except `createOptions` also has a relay. there may be an opportunity to pull these out, but not sure about it yet.

Reply:
yeah same thought crossed my mind and figured we could take a look after this pr lands

Reply:
https://github.com/TBD54566975/web5-go/issues/68